### PR TITLE
chore(form-core): fix prefilled FormGroups

### PR DESCRIPTION
**Pull request by tlouisse**
see https://github.com/ing-bank/lion/issues/1241

## What happened?

Assume this code:
`<lion-form .serializedValue="${{ x: 'y' }}">
      <lion-radio-group name="x">
        <lion-radio>`
        
`lionForm.dirty` became `true` on init, whereas it should be `false`...

This was because the initial `.serializedValue` was scheduled until after `registrationComplete`, just like `__initInteractonStates()`.
Since the latter was scheduled and executed first, the inital radio value caused the `dirty` state of the form to change.

## How to solve this?
- first, wait for registrationComplete to resolve.
- then resolve `initComplete` that guarantees all initial logic affecting children has run. The `__initInteractonStates` can wait for this Promise and everything works as expected.

## More changes in this mr:
- all registration logic is put in one mixin: FormRegistrarMixin
- ChoiceGroupMixin should not have InteractionStateMixin; it was designed to be compatible with Fields, and ChoiceGroupMixin can be applied on top of both Fields (listbox) and FormGroups (radio/cb-group)

### CODE REVIEW

File: packages/form-integrations/test/form-integrations.test.js [hunk 1]
```diff
@@ -37,23 +37,51 @@ describe('Form Integrations', () => {
...
+          .serializedValue="${{
+            full_name: { first_name: '', last_name: '' },
+            date: '2000-12-12',
+            datepicker: '2020-12-12',
+            bio: '',
+            money: '',
+            iban: '',
+            email: '',
+            checkers: ['foo', 'bar'],
+            dinosaurs: 'brontosaurus',
```
Comment (tlouisse) [inline packages/form-integrations/test/form-integrations.test.js:72]: The prefilled value for 'brontosaurus' currently breaks the test. ('' results in a value of `false` for dirty).
Obviously, solution should be looked for in radioGroup

File: packages/form-core/src/choice-group/ChoiceGroupMixin.js [hunk 2]
```diff
@@ -53,8 +52,8 @@ const ChoiceGroupMixinImplementation = superclass =>
       };
 
       if (this.__isInitialModelValue) {
-        this.__isInitialModelValue = false;
         this.registrationComplete.then(() => {
+          this.__isInitialModelValue = false;
```
Comment (tlouisse) [inline packages/form-core/src/choice-group/ChoiceGroupMixin.js:56]: These changes will prevent merge conflicts with https://github.com/ing-bank/lion/pull/1312

File: packages/form-core/src/form-group/FormGroupMixin.js [hunk 3]
```diff
@@ -146,32 +146,33 @@ const FormGroupMixinImplementation = superclass =>
...
-        },
-        () => {
-          this.registrationComplete.done = true;
-          throw new Error(
-            'Registration could not finish. Please use await el.registrationComplete;',
-          );
-        },
-      );
+
+      // TODO: shouldn't we put all this registration logic in FormRegistrarMixin?
```
Comment (tlouisse) [inline packages/form-core/src/form-group/FormGroupMixin.js:150]: comments will be deleted...

File: packages/form-integrations/test/helpers/umbrella-form.js [hunk 4]
```diff
@@ -23,9 +25,16 @@ export class UmbrellaForm extends LitElement {
...
+  set serializedValue(v) {
+    this.__serializedValue = v;
+  }
+
   render() {
     return html`
-      <lion-form>
+      <lion-form .serializedValue="${this.__serializedValue}">
         <form>
           <lion-fieldset name="full_name">
```
Comment (tlouisse) [inline packages/form-integrations/test/helpers/umbrella-form.js:39]: I took the same umbrell-form as in the docs, so that it contains all our FormControls

**Review by jorenbroekema — APPROVED** [review]
Looks cleaner having all this logic in the FormRegistrarMixin

**Pull request merged**